src/UserManager.ts

```diff
@@ -641,8 +654,9 @@ export class UserManager {
         const signoutResponse = await this._client.processSignoutResponse(url);
         logger.debug("got signout response");
 
-        await this.removeUser();
-        logger.debug("user removed");
+        if (!this.settings.raiserUserUnloadEventBeforeSignoutRequest) {
```

Comment:
maybe we should just call `await this._removeUser(true)` here, it should not hurt and is simpler (less code)

Reply:
I think this should be ok as a signed in user would be unexpected in any case, right?